### PR TITLE
feat: pool-detail-pair-info

### PR DESCRIPTION
**Pull request by ChefJerry**
---

## PR-Codex overview
This PR introduces changes related to token pairs, TVL, volume, and fees in various components and states.

### Detailed summary
- Added currency formatting to numbers
- Updated translations in `translations.json`
- Added new fields to `farmsV4` state and utils
- Updated components in `PoolDetail` view to display pair info, pool status, and tokens
- Added new components for pool tokens, currencies, and status

> The following files were skipped due to too many changes: `apps/web/src/views/PoolDetail/components/PoolCurrencies.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/views/PoolDetail/components/PoolStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    return new Percent(
+      BigInt((Number(poolInfo.tvlUsd ?? 0) - Number(poolInfo.tvlUsd24h ?? 0)).toFixed(0)),
+      BigInt(Number(poolInfo.tvlUsd ?? 1).toFixed(0)),
+    )
+  }, [poolInfo])
+  const volChange = useMemo(() => {
+    if (!poolInfo) return null
+    return new Percent(
+      BigInt((Number(poolInfo.vol24hUsd ?? 0) - Number(poolInfo.vol48hUsd ?? 0)).toFixed(0)),
+      BigInt(Number(poolInfo.vol24hUsd ?? 1).toFixed(0)) || BigInt(1),
```
Comment (chef-eric) [inline apps/web/src/views/PoolDetail/components/PoolStatus.tsx:28]: I think the calculator have some mistake, percent = (val(after) - val(before))/val(before), so the denominator should be vol48hUsd?

File: apps/web/src/views/PoolDetail/components/PoolTokens.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+export const PoolTokens: React.FC<PoolTokensProps> = ({ poolInfo }) => {
+  const chainName = useChainNameByQuery()
+  const [token0Tvl, token1Tvl] = useMemo(() => {
+    if (!poolInfo) return [0, 0]
+    return [
+      formatAmount(Number(poolInfo.tvlToken0 ?? 0), formatOptions),
+      formatAmount(Number(poolInfo.tvlToken1 ?? 0), formatOptions),
+    ]
+  }, [poolInfo])
```
Comment (chef-eric) [inline apps/web/src/views/PoolDetail/components/PoolTokens.tsx:34]: use poolInfo.tvlToken0 & poolInfo.tvlToken1 as deps will be better to reduce reCalc

File: apps/web/src/views/PoolDetail/components/PoolStatus.tsx (hunk 1 above)
Comment (chef-eric) [reply apps/web/src/views/PoolDetail/components/PoolStatus.tsx:28]: i found a existing util to do that, can use pancake-frontend/apps/web/src/utils/infoDataHelpers.ts:getPercentChange

File: apps/web/src/views/PoolDetail/components/ChangePercent.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  background: ${({ theme, $negative }) => ($negative ? theme.colors.destructive10 : theme.colors.positive10)};
+  border-radius: 999px;
+  padding: 0 4px;
+`
+
+type ChangePercentProps = {
+  percent: Percent
+}
+
+const IconUp = () => {
```
[the remark this answers is not in the document]
Comment (ChefJerry) [reply apps/web/src/views/PoolDetail/components/ChangePercent.tsx:16]: we can, will do it after confirm it is a global used with designer

File: apps/web/src/views/PoolDetail/components/PoolCurrencies.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  }
+`
+
+type PoolCurrenciesProps = {
+  poolInfo?: PoolInfo | null
+}
+export const PoolCurrencies: React.FC<PoolCurrenciesProps> = ({ poolInfo }) => {
+  const chainPath = useMultiChainPath()
+  const chainName = useChainNameByQuery()
+  const stableSwapPath = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (ChefJerry) [reply apps/web/src/views/PoolDetail/components/PoolCurrencies.tsx:26]: sure, just follow the old name

File: apps/web/src/views/PoolDetail/components/PoolStatus.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+type PoolStatusProps = {
+  poolInfo?: PoolInfo | null
+}
+export const PoolStatus: React.FC<PoolStatusProps> = ({ poolInfo }) => {
+  const { t } = useTranslation()
+  const tvlChange = useMemo(() => {
+    if (!poolInfo) return null
+    return new Percent(
+      BigInt((Number(poolInfo.tvlUsd ?? 0) - Number(poolInfo.tvlUsd24h ?? 0)).toFixed(0)),
+      BigInt(Number(poolInfo.tvlUsd ?? 1).toFixed(0)),
```
Comment (chef-eric) [inline apps/web/src/views/PoolDetail/components/PoolStatus.tsx:19]: maybe the same issue? pls check again

Comment (ChefJerry) [reply apps/web/src/views/PoolDetail/components/PoolStatus.tsx]: done

**Review by chef-eric — APPROVED** [review]

**Pull request merged**